components/BridgeForm.tsx

```diff
@@ -0,0 +1,223 @@
+import { BridgedEthereumToken, IntrinsicElements } from "@/types";
+import { FC, useCallback, useEffect, useState } from "react";
+import SubmitButton from "@/components/shared/SubmitButton";
+import { css } from "@emotion/react";
+import { Theme } from "@mui/material";
+import { useBridge } from "@/providers/BridgeProvider";
+import useBridgeStatus from "@/hooks/useBridgeStatus";
+import { useCENNZApi } from "@/providers/CENNZApiProvider";
+import { useMetaMaskWallet } from "@/providers/MetaMaskWalletProvider";
+import {
+	Balance,
+	ensureBridgeDepositActive,
+	ensureBridgeWithdrawActive,
+	sendDepositRequest,
+	sendWithdrawCENNZRequest,
+	ensureRelayerDepositDone,
+	ensureEthereumChain,
+} from "@/utils";
+import { useCENNZWallet } from "@/providers/CENNZWalletProvider";
+import sendWithdrawEthereumRequest from "@/utils/sendWithdrawEthereumRequest";
+import { useMetaMaskExtension } from "@/providers/MetaMaskExtenstionProvider";
+
+interface BridgeFormProps {}
+
+const BridgeForm: FC<IntrinsicElements["form"] & BridgeFormProps> = ({
+	children,
+	...props
+}) => {
+	const { api } = useCENNZApi();
+	const { wallet: metaMaskWallet } = useMetaMaskWallet();
+	const {
+		bridgeAction,
+		transferInput,
+		transferAsset,
+		transferCENNZAddress,
+		setProgressStatus,
+		setSuccessStatus,
+		setFailStatus,
+		setTxStatus,
+		updateMetaMaskBalances,
+		transferMetaMaskAddress,
+	} = useBridge();
+	const [buttonLabel, setButtonLabel] = useState<string>("Deposit");
+	const {
+		updateBalances: updateCENNZBalances,
+		selectedAccount: cennzAccount,
+		wallet: cennzWallet,
+	} = useCENNZWallet();
+	const { extension } = useMetaMaskExtension();
+
+	const processDepositRequest = useCallback(async () => {
+		const setTrValue = transferInput.setValue;
+		const transferAmount = Balance.fromInput(
+			transferInput.value,
+			transferAsset
+		);
+		setProgressStatus();
+
+		let tx: Awaited<ReturnType<typeof sendDepositRequest>>;
```

Comment:
ethers has a TransactionResponse type `import { TransactionResponse } from "@ethersproject/abstract-provider";`

Reply:
awesome, i've been looking for this